File: 766. Toeplitz Matrix/isToeplitzMatrix.js
/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var isToeplitzMatrix = function(matrix) {
    [row, column] = [matrix.length, matrix[0].length]
    for (let r=0;r<row-1;r++) {
        for (let c=0;c<column-1;c++) {
            if (matrix[r][c] != matrix[r+1][c+1]) {
                return false
            }
        }
    }
    return true
};

// Runtime 108 ms / Memory 44.2 MB
// debug: https://jsfiddle.net/

File: 766. Toeplitz Matrix/isToeplitzMatrix.cpp
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        int row=matrix.size(), col=matrix[0].size();
        for (int i=1; i<row; i++) {
            for (int j=1; j<col; j++) {
                // top-left to bottom-right
                if (matrix[i][j]!=matrix[i-1][j-1]) { return false; }
            }
        }
        return true;
    }
};